### PR TITLE
Webpack

### PR DESCRIPTION
**Pull request by whattokingu**
+ Changed task runner to Gulp (`npm test` runs `gulp test` rather than `make test`).
+ Added Webpack (via Gulp) as a build system.
+ Added Babel to both Browserify build and Webpack build.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -4,34 +4,39 @@ REPORTER = dot
...
+	# fi
 
 test-cov:
-	@./node_modules/.bin/istanbul cover ./node_modules/.bin/_mocha -- \
-		--reporter $(REPORTER) \
-		test/
+	@./node_modules/.bin/gulp test-cov
+	# @./node_modules/.bin/istanbul cover ./node_modules/.bin/_mocha -- \
+	# 	--reporter $(REPORTER) \
+	# 	test/
```
Comment (nkzawa) [inline Makefile]: We can just remove these comments ?

File: package.json [hunk 2]
```diff
@@ -27,22 +27,32 @@
     "backo2": "1.0.2"
   },
   "devDependencies": {
+    "babel-core": "6.4.5",
+    "babel-loader": "6.2.1",
+    "babel-preset-es2015": "6.3.13",
+    "babel-preset-react": "6.3.13",
```
Comment (nkzawa) [inline package.json]: I think `react` wouldn't be required.

**Comment by nkzawa** [comment]
:+1: basically, looks great to me

File: package.json (hunk 2 above)
Comment (paradite) [inline package.json]: will remove it

**Pull request merged**